tests/devtools.test.tsx

```diff
@@ -78,26 +78,45 @@ describe('If there is no extension installed...', () => {
     expect(console.warn).not.toBeCalled()
   })
 })
-
 describe('When state changes...', () => {
-  it("sends { type: setStateName || 'anonymous` } as the action with current state", () => {
-    const api = create(
-      devtools(() => ({ count: 0, foo: 'bar' }), {
-        name: 'testOptionsName',
-        enabled: true,
-      })
-    )
-    api.setState({ count: 10 }, false, 'testSetStateName')
+  it("sends { type: 'testSetStateName' } as the action with current state", () => {
+    const { setState } = createTestState()
+    setState({ count: 10 }, false, 'testSetStateName')
     expect(extension.send).toHaveBeenLastCalledWith(
       { type: 'testSetStateName' },
       { count: 10, foo: 'bar' }
     )
-    api.setState({ count: 5, foo: 'baz' }, true)
+  })
+
+  it("sends { type: 'testSetStateName', payload: 15 } as the action with current state", () => {
+    const { setState } = createTestState()
+    setState({ count: 15 }, false, {
+      type: 'testSetStateName',
+      payload: 15,
+    })
+    expect(extension.send).toHaveBeenLastCalledWith(
+      { type: 'testSetStateName', payload: 15 },
+      { count: 15, foo: 'bar' }
+    )
+  })
+
+  it("sends { type: 'anonymous' } as the action with current state", () => {
+    const { setState } = createTestState()
+    setState({ count: 5, foo: 'baz' }, true)
     expect(extension.send).toHaveBeenLastCalledWith(
       { type: 'anonymous' },
       { count: 5, foo: 'baz' }
     )
   })
+
+  function createTestState() {
+    return create(
+      devtools(() => ({ count: 0, foo: 'bar' }), {
+        name: 'testOptionsName',
+        enabled: true,
+      })
+    )
+  }
```

Comment:
I think this refactor is a quite unncessary, the only change required should be this...

```diff
  describe('When state changes...', () => {
-   it("sends { type: setStateName || 'anonymous` } as the action with current state", () => {
+   it("sends { type: setStateName || 'anonymous`, ...rest } as the action with current state", () => {
      const api = create(
        devtools(() => ({ count: 0, foo: 'bar' }), {
          name: 'testOptionsName',
          enabled: true,
        })
      )
      api.setState({ count: 10 }, false, 'testSetStateName')
      expect(extension.send).toHaveBeenLastCalledWith(
        { type: 'testSetStateName' },
        { count: 10, foo: 'bar' }
      )
+
+     api.setState({ count: 10 }, false, { type: 'testSetStateName', bar: 'foo' })
+     expect(extension.send).toHaveBeenLastCalledWith(
+       { type: 'testSetStateName', bar: 'foo' },
+       { count: 10, foo: 'bar' }
+     )
+
      api.setState({ count: 5, foo: 'baz' }, true)
      expect(extension.send).toHaveBeenLastCalledWith(
        { type: 'anonymous' },
        { count: 5, foo: 'baz' }
      )
    })
  })
```

Reply:
Does acting after assertion isn't a bad idea? It can lead to unexpected results. I think we should start with a fresh state and test one thing per `it`.

Reply:
It doesn't matter much nor do current tests follow such rule. So I think it's better to keep it minimal, at least for this PR.

I usually don't write my tests with that rule, but if @dai-shi wants prefers to have such rule then you can open another PR that refactors all tests to follow the rule and not just this one.

Reply:
You are right